db/migrate/20220920172912_devise_token_auth_create_users.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class DeviseTokenAuthCreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table(:users) do |t|
+      ## Required
+      t.string :provider, null: false, default: 'email'
+      t.string :uid, null: false, default: ''
+
+      ## Database authenticatable
+      t.string :encrypted_password, null: false, default: ''
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+      t.boolean  :allow_password_change, default: false
+
+      ## Confirmable
+      t.string   :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## User Info
+      t.string :email
```

Comment:
```suggestion
      t.string :email, null: false
```

And we should also add model validations that match these DB constrains